compiler/src/yul/mappers/declarations.rs

```diff
@@ -126,6 +127,9 @@ mod tests {
             }),
         );
 
-        assert_eq!(map(&harness.context, &harness.src), "let foo := alloc(200)");
+        assert_eq!(
+            map(&harness.context, &harness.src),
+            "let $foo := alloc(200)"
```

Comment:
Am I right to assume that all the runtime functions such as `alloc` should also turn into `$$alloc` (or maybe some scheme that sets them apart from user defined functions) to ensure those aren't callable by user code?

Reply:
Anytime someone writes a call expressions (e.g. `foo(42)`), the function identifier will include `$$` after being mapped (`$$foo(42)`), so it should be impossible for someone to write something that calls `alloc` or any other runtime function directly.

Reply:
Ah, sure! :see_no_evil: